### PR TITLE
Profile request + tiny bug correction

### PR DESCRIPTION
**Pull request by telegrapher**
## Description

Add ims/profile.go to handle profile requests to /ims/profile/v1

That endpoint returns an empty body in case of error 401, so I added a small condition to avoid decoding the empty body. With that I get at least the error code that is more valuable than the JSON decode error.

Warnings:

- I'm sending the PR for early feedback.
- My changes break 3 tests probably due to the small change in error handling, I will have a look tomorrow at that. 
- I didn't create any test for the profile.go, but it doesn't seem I can test much there, since it is extremely simple code.
- The function deals with simple strings, I avoid creating new struct types to hold only one string, but I can do that if you thing code homogeneity is more important than simplicity.

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ims/profile.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+package ims
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// The user token is used to authorize the request and to define which user's profile is requested.
+func (c *Client) GetProfile(userToken string) (string, error){
```
Comment (francescomari) [inline ims/profile.go:20]: I know this might sound silly, but I prefer to have a struct for the request and the response of this API call.

A struct for the request would allow the library to add new input parameters (if any are introduced in the future) without breaking the users of the library.

A struct for the response would allow the library to parse the user profile (which you currently don't do) without breaking the users of the library. Probably figuring out which fields to parse is a bit too much research for now (I know the status of the documentation). For the moment, though, you can just return a response struct with the body.

In summary, it would be nice if you could change the API to:

```go
type GetProfileRequest struct {
	AccessToken string
}

type GetProfileResponse struct {
	Body []byte
}

func (c *Client) GetProfile(req *GetProfileRequest) (*GetProfileResponse, error) {
	...
}
```

**Review by francescomari — CHANGES_REQUESTED** [review]

File: ims/error.go [hunk 2]
```diff
@@ -48,8 +49,14 @@ func errorResponse(r *http.Response) error {
...
 	}
 
-	if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
-		return fmt.Errorf("decode error response: %v", err)
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return fmt.Errorf("error reading request body")
+	}
+	if len(body) != 0 {
+		if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
```
[the remark this answers is not in the document]
Comment (francescomari) [reply ims/error.go:57]: This seems to be the reason for test failures in this PR.

File: ims/error.go [hunk 3]
```diff
@@ -48,8 +49,14 @@ func errorResponse(r *http.Response) error {
 		ErrorMessage string `json:"error_description"`
 	}
 
-	if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
-		return fmt.Errorf("decode error response: %v", err)
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return fmt.Errorf("error reading request body")
```
[the remark this answers is not in the document]
Comment (telegrapher) [reply ims/error.go]: Done

File: ims/error.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (telegrapher) [reply ims/error.go]: Done

File: ims/profile.go (hunk 1 above)
Comment (telegrapher) [reply ims/profile.go]: And done

**Review by francescomari — APPROVED** [review]

**Comment by francescomari** [comment]
There is some minor nitpicking, but I will take care of it myself before cutting the next release. Thanks for the contribution!
